core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -0,0 +1,414 @@
+                        auto tx = std::make_shared<RippleLikeBlockchainExplorerTransaction>();
+                        soci::session sql(self->getWallet()->getDatabase()->getPool());
+                        if (!RippleLikeTransactionDatabaseHelper::getTransactionByHash(sql, hash, *tx)) {
+                            throw make_exception(api::ErrorCode::TRANSACTION_NOT_FOUND, "Transaction {} not found",
+                                                 hash);
+                        }
+                        return tx;
+                    });
+        }
+
+        void RippleLikeAccount::inflateOperation(Operation &out,
+                                                 const std::shared_ptr<const AbstractWallet> &wallet,
+                                                 const RippleLikeBlockchainExplorerTransaction &tx) {
+            out.accountUid = getAccountUid();
+            out.block = tx.block;
+            out.rippleTransaction = Option<RippleLikeBlockchainExplorerTransaction>(tx);
+            out.currencyName = getWallet()->getCurrency().name;
+            out.walletType = getWalletType();
+            out.walletUid = wallet->getWalletUid();
+            out.date = tx.receivedAt;
+            if (out.block.nonEmpty())
+                out.block.getValue().currencyName = wallet->getCurrency().name;
+            out.rippleTransaction.getValue().block = out.block;
+        }
+
+        int RippleLikeAccount::putTransaction(soci::session &sql,
+                                              const RippleLikeBlockchainExplorerTransaction &transaction) {
+            auto wallet = getWallet();
+            if (wallet == nullptr) {
+                throw Exception(api::ErrorCode::RUNTIME_ERROR, "Wallet reference is dead.");
+            }
+
+            if (transaction.block.nonEmpty())
+                putBlock(sql, transaction.block.getValue());
+
+            int result = FLAG_TRANSACTION_UPDATED;
+
+            Operation operation;
+            inflateOperation(operation, wallet, transaction);
+            std::vector<std::string> senders{transaction.sender};
+            operation.senders = std::move(senders);
+            std::vector<std::string> receivers{transaction.receiver};
+            operation.recipients = std::move(receivers);
+            operation.fees = transaction.fees;
+            operation.trust = std::make_shared<TrustIndicator>();
+            operation.date = transaction.receivedAt;
+
+
+            if (_accountAddress == transaction.sender) {
+                operation.amount = transaction.value;
+                operation.type = api::OperationType::SEND;
+                operation.refreshUid();
+                if (OperationDatabaseHelper::putOperation(sql, operation)) {
+                    emitNewOperationEvent(operation);
+                }
+                result = FLAG_NEW_TRANSACTION;
+            }
+
+            if (_accountAddress == transaction.receiver) {
```

Comment:
You could use `else if` here to prevent a non-needed branching. Also, question: what happens if none of the branches are active? Is it possible in theory? If not, then we might want to return something indicating that the transaction is ill-formed?

Reply:
This is done in purpose, you can go through both branches (you send to yourself) or none of them (if transaction does not belong to you)